01 API Tutorials/04 Using Options in QuantConnect/04 Select Contracts.html

```diff
@@ -160,18 +160,18 @@
 
 <pre class="python">for i in slice.OptionChains:
     if i.Key != self.symbol: continue
-    chain = i.Value
-# differentiate the call and put options
-call = [x for x in optionchain if chain.Right == 0]
-put = [x for x in optionchain if chain.Right == 1]
-# choose ITM contracts
-contracts = [x for x in call if call.UnderlyingLastPrice - x.Strike &gt; 0]
-# or choose ATM contracts
-contracts = sorted(optionchain, key = lambda x: abs(optionchain.Underlying.Price - x.Strike))[0]
-# or choose OTM contracts
-contracts = [x for x in call if call.UnderlyingLastPrice - x.Strike &lt; 0]
-# sort the contracts by their expiration dates
-contracts = sorted(contracts, key = lambda x:x.Expiry, reverse = True)
+    optionchain = i.Value
+    # differentiate the call and put options
+    call = [x for x in optionchain if x.Right == 0]
+    put = [x for x in optionchain if x.Right == 1]
+    # choose ITM contracts
+    contracts = [x for x in call if call.UnderlyingLastPrice - x.Strike &gt; 0]
+    # or choose ATM contracts
+    contracts = sorted(optionchain, key = lambda x: abs(optionchain.Underlying.Price - x.Strike))[0]
+    # or choose OTM contracts
+    contracts = [x for x in call if call.UnderlyingLastPrice - x.Strike &lt; 0]
+    # sort the contracts by their expiration dates
+    contracts = sorted(contracts, key = lambda x:x.Expiry, reverse = True)
```

Comment:
We should use the `OptionRight` enum instead of its integer value for clarity.
It is a good practice to save the underlying price value instead of evaluating its value from the instance of `OptionContract` or `Security`. 

```python
            # differentiate the call and put options
            call = [x for x in optionchain if x.Right == OptionRight.Call]
            put = [x for x in optionchain if x.Right == OptionRight.Put]
            
            price = optionchain.Underlying.Price
            # choose ITM call contracts
            contracts = [x for x in call if price-x.Strike > 0]
            # or choose ATM call contracts
            contracts = sorted(optionchain, key=lambda x: abs(price - x.Strike))[0]
            # or choose OTM call contracts
            contracts = [x for x in call if price-x.Strike < 0]
            # sort the contracts by their expiration dates
            contracts = sorted(contracts, key=lambda x: x.Expiry, reverse = True)
```

Reply:
For ATM contracts, we could keep the code similar to ITM and OTM and therefore predictable?

```python
contracts = [x for x in optionchain if price-x.Strike == 0]
```

Reply:
I've made the the proposed change in the code.